iroh-bytes/src/provider.rs

```diff
@@ -487,34 +488,50 @@ pub enum SentStatus {
     NotFound,
 }
 
-/// Send a
+/// Send a blob to the client.
 pub async fn send_blob<D: Map, W: AsyncStreamWriter>(
     db: &D,
-    name: Hash,
+    hash: Hash,
     ranges: &RangeSpec,
     mut writer: W,
 ) -> Result<(SentStatus, u64, SliceReaderStats)> {
-    match db.get(&name).await? {
+    match db.get(&hash).await? {
         Some(entry) => {
             let outboard = entry.outboard().await?;
             let size = outboard.tree().size();
             let mut file_reader = TrackingSliceReader::new(entry.data_reader().await?);
             writer.write(size.to_le_bytes().as_slice()).await?;
-            let res = encode_ranges_validated(
+            encode_ranges_validated(
                 &mut file_reader,
                 outboard,
                 &ranges.to_chunk_ranges(),
                 writer,
             )
-            .await;
-            debug!("done sending blob {} {:?}", name, res);
-            res?;
+            .await
+            .map_err(|e| encode_error_to_anyhow(e, &hash))?;
 
             Ok((SentStatus::Sent, size, file_reader.stats()))
         }
         _ => {
-            debug!("blob not found {}", hex::encode(name));
+            debug!("blob not found {}", hash.to_hex());
```

Comment:
Usually it's better to do these as tracing fields:
```suggestion
            debug!(hash = %hash.to_hex(), "blob not found");
```